hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/SqlPlanCacheTest.java

```diff
@@ -123,6 +129,27 @@ public void test_dmlCaching() {
         assertThat(planCache(instance()).size()).isZero();
     }
 
+    @Test
+    public void test_indexCaching() {
+        fillMapAndCreateIndex(instance(), "m", IndexType.SORTED, "__key.id");
+        createMapping("map", "m", "id", PersonId.class, "varchar");
+        sqlService.execute("SELECT * FROM map ORDER BY id");
+        assertThat(planCache(instance()).size()).isEqualTo(1);
+        SqlPlan firstPlan = planCache(instance()).getPlans().values().iterator().next();
+
+        removeAllIndexes(instance(), "m");
+
+        fillMapAndCreateIndex(instance(), "m2", IndexType.HASH, "__key.id");
+        createMapping("map", "m2", "id", PersonId.class, "varchar");
```

Comment:
This test passes just because you create a mapping with different backing `IMap`.
I think the fact that below one fails is a problem:
```
@Test
public void test_index() {
    IMap<Object, Object> map = instance().getMap("m");

    createMapping("map", map.getName(), "id", PersonId.class, "varchar");
    map.addIndex(new IndexConfig(IndexType.SORTED, "__key.id").setName(randomName()));
    sqlService.execute("SELECT * FROM map ORDER BY id");
    assertThat(planCache(instance()).size()).isEqualTo(1);

    getMapContainer(map).getIndexes().destroyIndexes();
    map.addIndex(new IndexConfig(IndexType.HASH, "__key.id").setName(randomName()));
    assertThat(planCache(instance()).size()).isZero();
}
```

Reply:
AFAIU there is no good API to remove index from map ... Well, thanks.